src/azure-cli/azure/cli/command_modules/servicefabric/_params.py

```diff
@@ -111,3 +113,108 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
+                   'for example: --application-parameters param1=value1 param2=value2')
+        c.argument('minimum_nodes', options_list=['--min-nodes', '--minimum-nodes'],
+                   help='Specify the minimum number of nodes where Service Fabric will reserve capacity for this application.')
+        c.argument('maximum_nodes', options_list=['--max-nodes', '--maximum-nodes'],
+                   help='Specify the maximum number of nodes on which to place an application.')
+
+    # service
+    with self.argument_context('sf service') as c:
+        c.argument('service_name', options_list=['--name', '--service-name'],
+                   help='Specify the name of the service. The application name must be a prefix of the service name, for example: appName~serviceName')
+
+    with self.argument_context('sf service create', validator=validate_create_service) as c:
+        c.argument('service_type',
+                   help='Specify the service type name of the application, it should exist in the application manifest.')
+        c.argument('application_name', options_list=['--application', '--application-name'],
+                   help='Specify the name of the service. The application name must be a prefix of the service name, for example: appName~serviceName')
+        c.argument('state', arg_type=get_enum_type(['stateless', 'stateful']), help='Specify if the service is stateless or stateful.')
+        c.argument('instance_count', help='Specify the instance count for the stateless service. If -1 is used, it means it will run on all the nodes.')
+        c.argument('min_replica_set_size', help='Specify the min replica set size for the stateful service.')
+        c.argument('target_replica_set_size', help='Specify the target replica set size for the stateful service.')
+        c.argument('default_move_cost', arg_type=get_enum_type(['Zero', 'Low', 'Medium', 'High']),
+                   help='Specify the default cost for a move. Higher costs make it less likely that the Cluster Resource Manager will move the replica when trying to balance the cluster.')
+        c.argument('partition_scheme', arg_type=get_enum_type(['singleton', 'uniformInt64', 'named']),
+                   help='Specify what partition scheme to use. '
+                   'Singleton partitions are typically used when the service does not require any additional routing. '
+                   'UniformInt64 means that each partition owns a range of int64 keys. '
+                   'Named is usually for services with data that can be bucketed, within a bounded set. Some common examples of data fields used as named partition keys would be regions, postal codes, customer groups, or other business boundaries.')
+
+
+# pylint: disable=protected-access
+# pylint: disable=too-few-public-methods
+class addAppParamsAction(argparse._AppendAction):
```

Comment:
key=value pair is built in function inside azure cli core. https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/commands/__init__.py#L101

Reply:
I'm using it as described here: https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#keyvalue-arguments